students/jyeon/user/views.py

```diff
@@ -1,2 +1,90 @@
+import json
+import re
+import bcrypt
+import jwt
+
+from django.views      import View
+from django.http       import JsonResponse
+from django.db.models  import Q
+from django.db         import IntegrityError
+
+from mysettings        import SECRET_KEY, ALGORYTHM
+from .models           import User
+
+class UserSignupView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            
+            account_regex      = re.compile('^[A-Za-z0-9\.+_-]+\@[A-Za-z0-9\._-]+\.[a-zA-Z]+$')
+            phone_number_regex = re.compile('^[0-9]{3}\-*[0-9]{3,4}\-*[0-9]{3,4}$')
+            
+            #uniqueness check(account,phone_number,nickname)
+            if User.objects.filter(
+                Q(account      = data['account'])|
+                Q(phone_number = data['phone_number'])|
+                Q(nickname     = data.get('nickname'))).exists():
+                return JsonResponse({'message': 'ACCOUNT or PHONE NUMBER or NICK NAME ALEADY EXISTS'}, status=400)
+
+            # account validation
+            if not account_regex.match(data['account']):
+                return JsonResponse({'message':'INPUT ERROR'}, status=400)
+
+            # password validation
+            if len(data['password']) < 8:
+                return JsonResponse({'message': 'INPUT ERROR'}, status=400)
+            
+            # password encrypt
+            password_encoded = data['password'].encode('utf-8')
+            salt             = bcrypt.gensalt()
+            password_hashed  = bcrypt.hashpw(password_encoded, salt)
+            password_decoded = password_hashed.decode('utf-8')
```

Comment:
이 부분은 다 따로따로 굳이 변수화할 것 없이 한번에 처리해주셔도 됩니다.
```suggestion
            password_hashed  = bcrypt.hashpw(password_encoded, bcrypt.gensalt()).decode('utf-8')
```

Reply:
수정 완료했습니다!